python/orchestrator/command_steering_service.py

```
@@ -0,0 +1,175 @@
+        self.__kill_event = self.__signal_manager.kill_event
+        # initialize the Communicator object for communication via Queues
+        self.__communicator = None
+        self.__logger.debug("C&S service initialized.")
+
+    @property
+    def is_registered_in_registry(self): return self.__is_registered
+
+    # def __get_steering_menu_item(self, command):
+    #     """
+    #     maps the steering command with menu item.
+
+    #     Parameters
+    #     ----------
+    #     command : SteeringCommands.Enum
+    #         steering command enum
+
+    #     Returns
+    #     ------
+    #     menu item: returns the corresponding menu item, if found.
+    #     None: if not found.
+    #     """
+    #     menu_item = self.__steering_menu_handler.get_menu_item(command)
+    #     self.__logger.debug(f'selected menu item: {menu_item}')
+    #     if menu_item != Response.ERROR:
+    #         return menu_item
+    #     else:
+    #         return None
+
+    def run(self):
+        # register with registry
+        if(self.__component_service_registry_manager.register(
+                    os.getpid(),  # id
+                    SERVICE_COMPONENT_CATEGORY.COMMAND_AND_SERVICE,  # category
+                    SERVICE_COMPONENT_CATEGORY.COMMAND_AND_SERVICE,  # name
+                    (self.__command_and_steering_service_in_queue,  # endpoint
+                     self.__command_and_steering_service_out_queue),
+                    SERVICE_COMPONENT_STATUS.UP,  # current status
+                    None) == Response.OK):  # current state
+            # indicate a successful registration
+            self.__is_registered.set()
+            self.__logger.debug('Command and steering service is registered.')
+        else:
+            # quit if not registered
+            self.__logger.critical('quitting! \
+                            command and steering service is not registered.')
+            signal.raise_signal(signal.SIGTERM)
```

I would raise an exception here. 
There is no way to recover from this situation.

---

Raised, handled and logged the exception in new commit. Yes, it can execute the workflow if fails to register with registry.